File: GenesisAndroid/src/main/java/com/emerchantpay/gateway/genesisandroid/api/internal/TransactionResult.java
package com.emerchantpay.gateway.genesisandroid.api.internal;

import com.emerchantpay.gateway.genesisandroid.api.models.Transaction;

import com.emerchantpay.gateway.genesisandroid.api.util.NodeWrapper;

public class TransactionResult<T>{

    private Transaction transaction;

    public TransactionResult(NodeWrapper node, Class<T> klass) {
        this.transaction = new Transaction(node);
    }

    public Transaction getTransaction() {
        return transaction;
    }
}

File: GenesisAndroid/src/main/java/com/emerchantpay/gateway/genesisandroid/api/internal/TransactionResult.kt
package com.emerchantpay.gateway.genesisandroid.api.internal

import com.emerchantpay.gateway.genesisandroid.api.models.Transaction

import com.emerchantpay.gateway.genesisandroid.api.util.NodeWrapper

class TransactionResult<T>(node: NodeWrapper, klass: Class<T>) {

    val transaction: Transaction = Transaction(node)

}
